KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/LiveStreamSettingsModule.java

```diff
@@ -0,0 +1,243 @@
+		}
+
+	}
+
+	class LiveStreamPacketizerListener implements ILiveStreamPacketizerActionNotify {
+
+		private IApplicationInstance appInstance = null;
+
+		public LiveStreamPacketizerListener(IApplicationInstance appInstance) {
+			logger.debug("creating new LiveStreamPacketizerListener");
+			this.appInstance = appInstance;
+		}
+
+		public void onLiveStreamPacketizerCreate(ILiveStreamPacketizer liveStreamPacketizer, String streamName) {
+			logger.debug("onLiveStreamPacketizerCreate. stream: " + streamName);
+			if (!(liveStreamPacketizer instanceof LiveStreamPacketizerCupertino))
+				return;
+
+			LiveStreamPacketizerCupertino cupertinoPacketizer = (LiveStreamPacketizerCupertino) liveStreamPacketizer;
+			IMediaStream stream = this.appInstance.getStreams().getStream(streamName);
+
+			logger.info("Create [" + streamName + "]: " + liveStreamPacketizer.getClass().getSimpleName());
+			cupertinoPacketizer.setDataHandler(new LiveStreamPacketizerDataHandler(cupertinoPacketizer, streamName));
+		}
+
+		public void onLiveStreamPacketizerDestroy(ILiveStreamPacketizer liveStreamPacketizer) {
+			String streamName = liveStreamPacketizer.getAndSetStartStream(null).getName();
+			logger.debug("(" + streamName + ") onLiveStreamPacketizerDestroy");
+		}
+
+		public void onLiveStreamPacketizerInit(ILiveStreamPacketizer liveStreamPacketizer, String streamName) {
+			logger.debug("(" + streamName + ") onLiveStreamPacketizerInit");
+		}
+	}
+
+	public void onAppStart(IApplicationInstance applicationInstance) {
+		this.liveStreamPacketizerListener = new LiveStreamPacketizerListener(applicationInstance);
+		applicationInstance.addLiveStreamPacketizerListener(this.liveStreamPacketizerListener);
+		this.maxAllowedPTSDriftMillisec = applicationInstance.getProperties().getPropertyInt(MAX_ALLOWED_PTS_DRIFT_MILLISEC, DEFAULT_MAX_ALLOWED_PTS_DRIFT_MILLISEC);
+		logger.info("Modules.LiveStreamSettingsModule application instance started");
+	}
+
+	public void onStreamCreate(IMediaStream stream) {
+		stream.addLivePacketListener(new PacketListener());
+	}
+
+	public void onStreamDestroy(IMediaStream stream) {
+	}
+
+	class PacketListener implements IMediaStreamLivePacketNotify {
+
+		private long lastPacketTimeCode;
+		private long baseSystemTime;
+		private long basePacketTimeCode;
+		private String entryId = null;
+
+		public PacketListener() {
+			this.lastPacketTimeCode = 0;
+			this.basePacketTimeCode = 0;
+			this.baseSystemTime = 0;
+		}
+
+		public void onLivePacket(IMediaStream stream, AMFPacket thisPacket) {
+
+			long currentTime = System.currentTimeMillis();
+			long currentPacketTimeCode = thisPacket.getAbsTimecode();
+			long currentPTS = thisPacket.getTimecode();
+			boolean isAudio = thisPacket.isAudio();
+			boolean isVideo = thisPacket.isVideo();
+			String streamName = stream.getName();
+
+			if (this.entryId == null) {
```

Comment:
removed the duplicate function. Now using Utils.getEntryIdFromStreamName(streamName)